presto-main/src/main/java/com/facebook/presto/operator/scalar/JsonOperators.java

```diff
@@ -322,11 +325,40 @@ public static Slice castFromVarchar(@SqlType("varchar(x)") Slice value)
 
     @ScalarOperator(CAST)
     @SqlType(JSON)
+    public static Slice castFromTinyInt(@SqlType(TINYINT) long value)
+            throws IOException
+    {
+        return internalCastFromLong(value, 8);
+    }
+
+    @ScalarOperator(CAST)
+    @SqlType(JSON)
+    public static Slice castFromSmallInt(@SqlType(SMALLINT) long value)
+            throws IOException
+    {
+        return internalCastFromLong(value, 12);
+    }
+
+    @ScalarOperator(CAST)
+    @SqlType(JSON)
     public static Slice castFromInteger(@SqlType(INTEGER) long value)
             throws IOException
     {
+        return internalCastFromLong(value, 20);
```

Comment:
A more accurate estimate for tinyint/smallint/int seems to be 4/6/11. Do you think this is correct?